python_modules/dagster/dagster/_core/pipes/utils.py

```diff
@@ -332,7 +327,18 @@ def get_params(self) -> Iterator[PipesParams]:
         """
 
     @abstractmethod
-    def download_messages_chunk(self, index: int, params: PipesParams) -> Optional[str]: ...
+    def download_messages(
+        self, cursor: Optional[TCursor], params: PipesParams
+    ) -> Optional[Tuple[TCursor, str]]:
+        """Download a chunk of messages from the target location.
+
+        Args:
+            cursor (Optional[Any]): A cursor that specifies where to start downloading messages from.
+                it can be set according to specifics of each message reader implementation, for example.
+                it can be an index for a line in a log file, or a timestamp for a message in a stream.
+            params (PipesParams): A dict of parameters that specifies where to download messages from.
```

Comment:
```
cursor (Optional[Any]): Cursor specifying start location from which to download 
    messages in a stream. The format of the value varies with the message reader 
    implementation. It might be an integer index for a line in a log file, or a 
    timestamp for a message in a time-indexed stream.
```